In [17]:
results = {
    'vk': {'revenue': 103, 'cost': 98},
    'yandex': {'revenue': 179, 'cost': 153},
    'facebook': {'revenue': 103, 'cost': 110},
    'adwords': {'revenue': 35, 'cost': 34},
    'twitter': {'revenue': 11, 'cost': 24},
}
for i, j in results.items():
    revenue = j['revenue']
    cost = j['cost']
    ROI = ((revenue / cost) - 1) * 100
    results[i]['ROI'] = round(ROI, 3)
print(f'{results}')

{'vk': {'revenue': 103, 'cost': 98, 'ROI': 5.102}, 'yandex': {'revenue': 179, 'cost': 153, 'ROI': 16.993}, 'facebook': {'revenue': 103, 'cost': 110, 'ROI': -6.364}, 'adwords': {'revenue': 35, 'cost': 34, 'ROI': 2.941}, 'twitter': {'revenue': 11, 'cost': 24, 'ROI': -54.167}}


Дана переменная, в которой хранится информация о затратах и доходе рекламных кампаний по различным источникам. Необходимо дополнить исходную структуру показателем ROI, который рассчитаем по формуле: ((revenue / cost) - 1) * 100

Пример работы программы:

results = {
    'vk': {'revenue': 103, 'cost': 98},
    'yandex': {'revenue': 179, 'cost': 153},
    'facebook': {'revenue': 103, 'cost': 110},
    'adwords': {'revenue': 35, 'cost': 34},
    'twitter': {'revenue': 11, 'cost': 24},
}
Результат:

{'adwords': {'revenue': 35, 'cost': 34, 'ROI': 2.94},
 'facebook': {'revenue': 103, 'cost': 110, 'ROI': -6.36},
 'twitter': {'revenue': 11, 'cost': 24, 'ROI': -54.17},
 'vk': {'revenue': 103, 'cost': 98, 'ROI': 5.1},
 'yandex': {'revenue': 179, 'cost': 153, 'ROI': 16.99}}